app/controllers/password_controller.rb

```diff
@@ -0,0 +1,86 @@
+require 'auth/authentication_backend'
+
+class PasswordController < ApplicationController
+  include AuthenticationBackend
+
+  skip_before_action :authenticate_user!, except: %w[password_form update_password]
+  skip_before_action :set_user, except: %w[password_form update_password]
+
+  layout 'main_layout', only: :password_form
+
+  def password_form
+    @password_form = ChangePasswordForm.new({})
+  end
+
+  def update_password
+    @password_form = ChangePasswordForm.new(params[:change_password_form])
+    if @password_form.valid?
+      change_password(
+        current_password: @password_form.old_password,
+        new_password: @password_form.password,
+        access_token: current_user.access_token
+      )
+      flash[:notice] = t('password.password_changed')
+      redirect_to profile_path
+    else
+      flash.now[:errors] = @password_form.errors.full_messages.join(', ')
+      render :password_form, status: :bad_request
+    end
+  rescue InvalidOldPasswordError
+    flash[:error] = t('password.errors.old_password_mismatch')
+    render :password_form, status: :bad_request
+  rescue InvalidNewPasswordError
+    flash[:error] = t('devise.sessions.InvalidPasswordException')
+    render :password_form, status: :bad_request
+  rescue AuthenticationBackendException
+    flash[:error] = t('devise.failure.unknown_cognito_response')
+    render :password_form, status: :bad_request
+  end
+
+  def forgot_form
+    @form = ForgottenPasswordForm.new({})
+  end
+
+  def send_code
+    @form = ForgottenPasswordForm.new(params[:forgotten_password_form])
+    if @form.valid?
+      request_password_reset(@form.to_h)
+      session[:email] = @form.email
+      @form = PasswordRecoveryForm.new({})
+      render :user_code
+    else
+      flash.now[:errors] = @form.errors.full_messages.join(', ')
+      render :forgot_form, status: :bad_request
+    end
+  rescue ToManyAttemptsError
+    flash.now[:errors] = t('password.errors.to_many_attempts')
+    render :forgot_form, status: :bad_request
```

Comment:
I don't think we should be rescuing and surfacing the error message to the user. For example, if they put a non-existing email it just blows up with an exception. We should just go to the next step no matter what. This prevents people from trying and fishing for registered/existing emails...

Reply:
same with when the user doesn't have a verified email in their profile... it blows up with an uncaught exception.. (which reminds me that when creating users we need to make sure their email is verified)

Reply:
Changed this.